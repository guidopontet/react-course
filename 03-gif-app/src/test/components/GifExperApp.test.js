const { shallow } = require("enzyme");

const { default: GifExperApp } = require("../../components/GifExperApp");

describe('GifExperApp', () => {
  test('should render component', () => {
    const wrapper = shallow(<GifExperApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render a list of categories', () => {
    const categories = ['Pokem√≥n', 'Dragon Ball'];

    const wrapper = shallow(<GifExperApp defaultCategories={ categories } />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
