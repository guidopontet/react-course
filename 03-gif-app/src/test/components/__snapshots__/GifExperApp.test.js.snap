// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExperApp should render a list of categories 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pokemón"
      key="Pokemón"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`GifExperApp should render component 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
